src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -9879,31 +9887,84 @@ export class StudyViewPageStore
     });
 
     @action
-    showAsPieChart(uniqueKey: string, dataSize: number): void {
+    showAsPieChart(
+        uniqueKey: string,
+        data: ClinicalDataCount[] | string[]
+    ): void {
+        const { totalCount, naCount } = (data as (
```

Comment:
add comments here to explain what's happening here. can also add comments below to describe the method of choosing which chart